admin-dev/themes/new-theme/js/components/grid/extension/column/catalog/category-position-extension.js

```diff
@@ -0,0 +1,171 @@
+    const way = (this.originalPositions.indexOf(row.id) < positions.indexOf(row.id)) ? 1 : 0;
+
+    const $categoryPositionContainer = $(row).find('.js-' + this.grid.getId() + '-position:first');
+
+    const categoryId = $categoryPositionContainer.data('id');
+    const categoryParentId = $categoryPositionContainer.data('id-parent');
+    const positionUpdateUrl = $categoryPositionContainer.data('position-update-url');
+
+    let params = positions.replace(new RegExp(this.grid.getId() + '_grid_table', 'g'), 'category');
+
+    let queryParams = {
+      id_category_parent: categoryParentId,
+      id_category_to_move: categoryId,
+      way: way,
+      ajax: 1,
+      action: 'updatePositions'
+    };
+
+    if (positions.indexOf('_0&') !== -1) {
+      queryParams.found_first = 1;
+    }
+
+    params += '&' + $.param(queryParams);
+
+    this._updateCategoryPosition(positionUpdateUrl, params);
+  }
+
+  /**
+   * Add ID's to Grid table rows to make tableDnD.onDrop() function work.
+   *
+   * @private
+   */
+  _addIdsToGridTableRows() {
+    this.grid.getContainer()
+      .find('.js-grid-table')
+      .find('.js-' + this.grid.getId() + '-position')
+      .each((index, positionWrapper) => {
+        const $positionWrapper = $(positionWrapper);
+
+        const categoryId = $positionWrapper.data('id');
+        const categoryParentId = $positionWrapper.data('id-parent');
+        const position = $positionWrapper.data('position');
+
+        const id = 'tr_' + categoryParentId + '_' + categoryId + '_' + position;
+
+        $positionWrapper.closest('tr').attr('id', id);
+      });
+  }
+
+  /**
+   * Update categories listing with new positions
+   *
+   * @private
+   */
+  _updateCategoryIdsAndPositions() {
+    this.grid.getContainer()
+      .find('.js-grid-table')
+      .find('.js-' + this.grid.getId() + '-position')
+      .each((index, positionWrapper) => {
+        const $positionWrapper = $(positionWrapper);
+        const $row = $positionWrapper.closest('tr');
+
+        const offset = $positionWrapper.data('pagination-offset');
+        const newPosition = offset > 0 ? index + offset : index;
+
+        const oldId = $row.attr('id');
+        $row.attr('id', oldId.replace(/_[0-9]$/g, '_' + newPosition));
+
+        $positionWrapper.find('.js-position').text(newPosition + 1);
+        $positionWrapper.data('position', newPosition);
+      });
+  }
+
+  /**
+   * Process categories positions update
+   *
+   * @param {String} url
+   * @param {String} params
+   *
+   * @private
+   */
+  _updateCategoryPosition(url, params) {
+    $.post({
+      url: url,
+      headers: {
+        'cache-control': 'no-cache'
+      },
+      data: params
+    }).then((response) => {
+      response = JSON.parse(response);
+
+      if (typeof response.message !== 'undefined') {
+        showSuccessMessage(response.message);
+      } else {
+        // use legacy error
+        // update when all category controller is migrated to symfony
```

Comment:
Can you add a `@todo` statement here to allow easy findings of such WIP parts of the code ?

Additionnaly it would be great to add a comment inside the related issue (you should be able to find it in https://github.com/PrestaShop/PrestaShop/projects/3) "when this controller is migrated, fix that !" ;)